### PR TITLE
Added Welcome Screen Notice Message

### PR DESCRIPTION
**Pull request by kbarkevich**
Filled-in the "ANNOUNCE" section to present the user with a notice welcoming them to the MH3SP Beta Test. The announcement reads as follows:
```
          ---------- MH3SP ----------
     Monster Hunter 3 (Tri) Server Project

Welcome to the MH3SP Beta Test! We ask that you
keep few things in mind while connected in order to
make the experience as fun as possible for everyone.
No abusing other players, no exploiting bugs, and
please report bugs that you find to the server
staff/dev team. We thank you for your patience as
we prepared this server to be fit for public use,
and ask your continued patience in the weeks to
come as you may encounter some bugs and errors
while playing. Please see the Public Test Server
Roadmap for a list of features we are working on.
             Welcome to Loc Lac!
```

### CODE REVIEW

**Review by InusualZ — APPROVED** [review]
LGTM

**Comment by kbarkevich** [comment]
Another small addition was pushed, adding city greeter text that changes based on whether or not the Sandstorm is active.

File: mh/constants.py [hunk 1]
```diff
@@ -164,13 +166,43 @@ def make_binary_npc_greeters(is_jap=False):
     US_OFFSET = 0x180
     JP_OFFSET = 0x100
     offset = JP_OFFSET if is_jap else US_OFFSET
+
+    day_in_cycle = int(current_server_time()//SECONDS_PER_DAY) % JHEN_EVENT_OFFSET
+    is_jhen_active = day_in_cycle >= JHEN_START and day_in_cycle < JHEN_END
```
Comment (InusualZ) [inline mh/constants.py:171]: You should create a helper method for detecting when Jhen is active

**Review by InusualZ — COMMENTED** [review]
Only one comment

File: mh/constants.py [hunk 2]
```diff
@@ -164,13 +166,43 @@ def make_binary_npc_greeters(is_jap=False):
...
-    data += pad(b"Material shop unavailable\nyet.", offset)
-    data += pad(b"Trading post closed at\nthe moment.", offset)
-    data += pad(b"No event quests.", offset)
+    data += pad(tool_shop_str.encode(), offset)
+    data += pad(material_shop_str.encode(), offset)
+    data += pad(b"The Trading Post is open!", offset)
+    data += pad(event_quests_str.encode(), offset)
     data += pad(b"No arena quests.", offset)
-    data += pad(b"To all hunters:\n\nThis is a test server.", offset)
+    data += pad(guildmaster_str.encode(), offset)
```
Comment (sepalani) [inline mh/constants.py:205]: Please make these additions by creating another PR.

File: mh/constants.py [hunk 3]
```diff
@@ -239,7 +271,22 @@ def slot(item, qty):
...
+                "No abusing other players, no exploiting bugs, and",
+                "please report bugs that you find to the server",
+                "staff/dev team. We thank you for your patience as",
+                "we prepared this server to be fit for public use,",
+                "and ask your continued patience in the weeks to",
+                "come as you may encounter some bugs and errors",
+                "while playing. Please see the Public Test Server",
+                "Roadmap for a list of features we are working on.",
+                "<CENTER>Welcome to Loc Lac!<END>"]
+ANNOUNCE = "<BR><BODY>".join(sum([[line] for line in ANNOUNCE], []))
```
Comment (sepalani) [inline mh/constants.py:288]: AFAICT, this is equivalent to: `"<BR><BODY>".join(ANNOUNCE)` so do this instead of redefining the variable _(keep also in mind that this variable is `bytes` not `str`)_:
```python
ANNOUNCE = b"<BR><BODY>".join([
    b"<BODY><CENTER>---------- MH3SP ----------",
    b"<BR><BODY>Monster Hunter 3 (Tri) Server Project",
    b"<BR><LEFT>Welcome to the MH3SP Beta Test! We ask that you",
    b"keep few things in mind while connected in order to",
    b"make the experience as fun as possible for everyone.",
    b"No abusing other players, no exploiting bugs, and",
    b"please report bugs that you find to the server",
    b"staff/dev team. We thank you for your patience as",
    b"we prepared this server to be fit for public use,",
    b"and ask your continued patience in the weeks to",
    b"come as you may encounter some bugs and errors",
    b"while playing. Please see the Public Test Server",
    b"Roadmap for a list of features we are working on.",
    b"<CENTER>Welcome to Loc Lac!<END>"
])
```

Also the message is too long and doesn't fit, see this screenshot from the Japanese version:
image

**Review by sepalani — CHANGES_REQUESTED** [review]

File: mh/constants.py (hunk 2 above)
Comment (kbarkevich) [reply mh/constants.py]: The title of this PR has been changed to be more accurate to what it's supposed to do

File: mh/constants.py (hunk 1 above)
Comment (kbarkevich) [reply mh/constants.py]: Done

File: mh/constants.py (hunk 3 above)
Comment (InusualZ) [reply mh/constants.py:285]: I think this can be work around by explicitly setting a font size

Comment (kbarkevich) [reply mh/constants.py:269]: This has been fixed by trmoeving the top "------ MH3SP ------" line in the Japanese version, since only a few characters were cut off.

Comment (sepalani) [reply mh/constants.py]: This doesn't look like a proper fix to me. Moreover, even if on the western version the text is displayed, it's way too close of the <kbd>a</kbd> button. The issue is still happening if a Japanese game connects to a western server since you're using the server `IS_JAP` config.

File: mh/constants.py [hunk 4]
```diff
@@ -248,6 +249,27 @@ def slot(item, qty):
...
+            "come as you may encounter some bugs and errors",
+            "while playing. Please see the Public Test Server",
+            "Roadmap for a list of features we are working on.",
+            "<CENTER>Welcome to Loc Lac!<END>"
+]
+if IS_JAP:
+    ANNOUNCE = ANNOUNCE[1:]
+    ANNOUNCE[0] = "<CENTER>" + ANNOUNCE[0]
+ANNOUNCE = "<BR><BODY>".join(ANNOUNCE)
+
```
Comment (sepalani) [inline mh/constants.py:272]:
```python
ANNOUNCE = b"<BR><BODY>".join([
    b"<BR><CENTER><BODY>Monster Hunter 3 (Tri) Server Project",
    b"<BR><LEFT>Welcome to the MH3SP Beta Test! We ask that you",
    b"keep few things in mind while connected in order to",
    b"make the experience as fun as possible for everyone.",
    b"No abusing other players, no exploiting bugs, and",
    b"please report bugs that you find to the server",
    b"staff/dev team. We thank you for your patience as",
    b"we prepared this server to be fit for public use,",
    b"and ask your continued patience in the weeks to",
    b"come as you may encounter some bugs and errors",
    b"while playing. Please see the Public Test Server",
    b"Roadmap for a list of features we are working on.",
    b"<BR><CENTER>Welcome to Loc Lac!<END>"
])
```

**Review by sepalani — CHANGES_REQUESTED** [review]

File: mh/constants.py (hunk 4 above)
Comment (kbarkevich) [reply mh/constants.py:272]: This would remove the logic that adds/removes the first line depending on whether or not it's in japanese mode.

Comment (sepalani) [reply mh/constants.py:272]: That's intended as this line doesn't provide much value and it does work for all versions.

**Comment by kbarkevich** [comment]
This PR is ALL just fluff/flavor text, almost none of it adds value. It's
just to make the design look nicer, which I think that first line does. And
with the logic I added to check IS_JAP, everything fits the way it should.
I don't see the problem here

**Comment by sepalani** [comment]
### Keep things simple (KISS principle)
> This PR is ALL just fluff/flavor text, almost none of it adds value.

1. The original code was one line, a simple `bytes` literal.
2. You replaced it with a bunch of code, and an overly complicated logic:
    - By checking the server config to avoid a bug/visual glitch rather than fixing it.
    - By adding more code and monkey patching the previous string so the issue doesn't appear.

There are almost no reason to add this much code if this PR doesn't add much value.

### Don't add superfluous code 
> It's just to make the design look nicer, which I think that first line does.

IMHO, the first line `---------- MH3SP ----------` is useless and redundant with the second line. So why should we keep it when it prevents the issue on all versions, is simple and visually acceptable?

I don't know if you tried to stick to a regional variant of the announce message but if you did, their wording and style (alignment, font size, etc.) are different depending on the language. In other word, it doesn't matter if they used `---` headings since some didn't.

### Try to address the root cause of the issue
> And with the logic I added to check IS_JAP, everything fits the way it should.

It's not. As I stated, a Japanese version of the game can still connect to a western server `IS_JAP=False` and faces this issue. It can connect to some extent depending on the HR _(HR0 hunters won't face connection issues)_. Regardless, it doesn't address the issue and checking the config flag is just an incomplete workaround. Trimming the unnecessary bits of the message is a cleaner and simpler workaround that's working everywhere.

### This project is open-source and driven by multiple persons
> I don't see the problem here

As I have already stated, your code/changes might be read/reviewed by other peers using this project. Obviously, it means that contributors should write clean code and not over-complicate problems without valid reasons. Adding this much logic whereas it can be expressed in a simpler/cleaner/more pythonic way **is a problem**.

_TL;DR: I won't be willing to merge this as long as my comment isn't addressed._

File: mh/constants.py (hunk 4 above)
Comment (sepalani) [reply mh/constants.py:272]: Here is a new suggestion using styles as suggested by @InusualZ 
```python
ANNOUNCE = b"<BR><BODY>".join([
    b"<BR><CENTER><BODY><SIZE=6>Monster Hunter 3 (Tri) Server Project",
    b"<BR><LEFT><SIZE=4>Welcome to the MH3SP Beta Test! We ask that you",
    b"keep few things in mind while connected in order to",
    b"make the experience as fun as possible for everyone.",
    b"No abusing other players, no exploiting bugs, and",
    b"please report bugs that you find to the server",
    b"staff/dev team. We thank you for your patience as",
    b"we prepared this server to be fit for public use,",
    b"and ask your continued patience in the weeks to",
    b"come as you may encounter some bugs and errors",
    b"while playing. Please see the Public Test Server",
    b"Roadmap for a list of features we are working on.",
    b"<BR><CENTER><C=2>Welcome to Loc Lac!<END>"
])
```
image

Some notes regarding the tag format:
 - it seems *(I haven't checked the code yet)* the game only picks the first digit after the equal sign *(i.e. `COLOR=1` is the same as `COLOR=10`)*
 - the `<C>` tag seems to be an alias to the `<COLOR>` tag

**Comment by sepalani** [comment]
I updated the PR, feel free to open a new one if there are adjustments to be made.
